### DISCUSSION TITLE
Custom dragging logic for component is not there

### DISCUSSION THREAD

**Discussion by RadheyMugdal** [GrapesJS/grapesjs#6165] [Q&A]
Actually in my project i am trying to add custom dragging logic for components inside canvas. I want that perticuler component can only be draggable and droppable inside its current  immediate parent but not able to do so.
` editor.on('component:drag:start',(e)=>{
      
      const parent=e.target.parent()
      editor.on('canvas:dragover',(e)=>{
        console.log(e);
        
      })
      //if component is being dragged over element other than immediate parent than
      e.target.set('draggable',false)
      editor.on('component:drag:end',(e)=>{
          
      })`

**Comment by mohamedsalem401** [comment]
Hey @RadheyMugdal, a simpler solution would be to use `draggable` and `droppable` properties for the components. You can check the API here https://grapesjs.com/docs/api/component.html#component
